### PR TITLE
fix: add build resource dir to classpath element to scan

### LINKED ISSUE #14420: Gradle multimodule project - production build fails due to static assets missing
### Description of the bug

Multimodule project - Subproject 1 + Subproject 2

Subproject 2 uses static assets from Submodule 1

./gradlew bootRun or ./gradlew bootJar works fine but when setting "-Pvaadin.productionMode=true" frontend compilation fails:

```
* What went wrong:
Execution failed for task ':theme:vaadinBuildFrontend'.
> Webpack process exited with non-zero exit code.

(...)

Module not found: Error: Can't resolve './styles/colors.css' in ' (--truncated--) Vaadin/project/theme/build/frontend'
```
This happens for every module in the resources dir.

### Expected behavior

Production and dev builds should both complete but only dev works.

### Minimal reproducible example

Subproject1 - Theme

build.gradle:

```
plugins {
    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
    id 'java'
    id 'com.vaadin'
}

group 'com.xxx.designsystem.vaadin'
version '1.0-SNAPSHOT'

sourceCompatibility = JavaVersion.VERSION_11
targetCompatibility = JavaVersion.VERSION_11

repositories {
    mavenCentral()
    maven { setUrl("https://maven.vaadin.com/vaadin-prereleases") }
    maven { setUrl("https://maven.vaadin.com/vaadin-addons") }
}

dependencyManagement {
    imports {
        mavenBom "com.vaadin:vaadin-bom:$vaadinVersion"
    }
}

dependencies {
    implementation 'com.vaadin.componentfactory:tooltip:2.0.2'
    implementation 'org.vaadin.gatanaso:multiselect-combo-box-flow:4.0.0-rc2'
}

test {
    useJUnitPlatform()
}
```

Subproject2 - Vaadin demo app

build.gradle:

```
plugins {
    id 'org.springframework.boot' version '2.5.12'
    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
    id 'java'
    id 'com.vaadin'
}

group 'com.xxx.designsystem.demo'
version '1.0-SNAPSHOT'

sourceCompatibility = JavaVersion.VERSION_11
targetCompatibility = JavaVersion.VERSION_11

repositories {
    mavenCentral()
    maven { setUrl("https://maven.vaadin.com/vaadin-prereleases") }
    maven { setUrl("https://maven.vaadin.com/vaadin-addons") }
}

configurations {
    developmentOnly
    runtimeClasspath {
        extendsFrom developmentOnly
    }
}

dependencyManagement {
    imports {
        mavenBom "com.vaadin:vaadin-bom:$vaadinVersion"
    }
}

dependencies {
    implementation project(":theme")
    implementation('com.vaadin:vaadin-spring-boot-starter')
    implementation 'com.vaadin.componentfactory:tooltip:2.0.2'
    implementation 'org.vaadin.gatanaso:multiselect-combo-box-flow:4.0.0-rc2'
    developmentOnly 'org.springframework.boot:spring-boot-devtools'
    testImplementation('org.springframework.boot:spring-boot-starter-test') {
        exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
    }
}

test {
    useJUnitPlatform()
}
```

Root project - build.gradle:
```
plugins {
    id 'java'
}

group 'com.xxx.designsystem'
version '1.0-SNAPSHOT'

repositories {
    mavenCentral()
    maven { setUrl("https://maven.vaadin.com/vaadin-prereleases") }
    maven { setUrl("https://maven.vaadin.com/vaadin-addons") }
}

dependencies {
    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.0'
    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
}

test {
    useJUnitPlatform()
}
```

Add static resources to theme/src/main/resources/META-INF/resources/whatever.css and build a component that uses it.
Then use that component in vaadin-demo-app (subproject2)

```
./gradlew bootRun -> works
./gradlew bootJar -> works
./gradlew "-Pvaadin.productionMode=true" bootRun -> fails
./gradlew "-Pvaadin.productionMode=true" bootRun -> fails
```

### Versions

Vaadin 23.1.6
'io.spring.dependency-management' version '1.0.11.RELEASE'
gradle: 6.8.3
'org.springframework.boot' version '2.5.12'

### PR DESCRIPTION
**Pull request by mcollovati**
## Description

ClassFinder used for frontend build is missing build resources directory. This change adds it to the list of elements to scan during build.

Fixes #14420

## Type of change

- [X] Bugfix
- [ ] Feature

## Checklist

- [X] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [X] I have added a description following the guideline.
- [X] The issue is created in the corresponding repository and I have referenced it.
- [X] I have added tests to ensure my change is effective and works as intended.
- [X] New and existing tests are passing locally with my change.
- [X] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Comment by mcollovati** [comment]
On cherry-picks, we should probably change the version of the `vaadin-themable-mixin` NPM package

```java
 @NpmPackage(value = "@vaadin/vaadin-themable-mixin", version = "23.3.0-alpha1")
```

File: flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/GradlePluginAdapter.kt [hunk 1]
```diff
@@ -53,14 +53,17 @@ internal class GradlePluginAdapter(val project: Project, private val isBeforePro
             .toList()
             .filter { it.exists() }
 
+        val resourcesDir: List<File> = listOfNotNull(sourceSet.getByName("main").output.resourcesDir)
```
Comment (mshabarov) [inline flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/GradlePluginAdapter.kt:56]: I couldn't find where `listOfNotNull` is defined?

Comment (mshabarov) [reply flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/GradlePluginAdapter.kt:56]: Is it Kotlin's native method?

Comment (mshabarov) [reply flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/GradlePluginAdapter.kt:56]: Yes, it is :)

**Review by mshabarov — APPROVED** [review]

**Comment by mshabarov** [comment]
I've added `target/9.0`, since V22 also has gradle plugin.

**Pull request merged**